### PR TITLE
Allow product to only provide one of the queries

### LINKED ISSUE #98: Allow users to provide either `blocker` or `critical-blocker` (or both) rather than requiring both
We have found that some teams only want to define one query, as there is no differentiation between `blocker` and `critical-blocker` in their workflow. This will just print a warning message that only one query is defined for that product, but will still run the query that is defined.

### PR DESCRIPTION
**Pull request by calebevans**
See issue for details. Making this change per a discussion in our weekly sync on 04/10.

Closes #98

### CODE REVIEW

File: qe_metrics/utils/general.py [hunk 1]
```diff
@@ -42,6 +42,13 @@ def verify_queries(queries_dict: dict[str, str]) -> None:
     """
     required_queries = ["blocker", "critical-blocker"]
 
+    missing_queries = [query for query in required_queries if query not in queries_dict]
+
+    if len(missing_queries) == len(required_queries):
+        raise ValueError(f"All queries are missing in the products file: {' '.join(missing_queries)}")
+    elif missing_queries:
+        LOGGER.warn(f"Some queries is missing in the products file: {' '.join(missing_queries)}")
```
Comment (rnetser) [inline qe_metrics/utils/general.py:50]:
```suggestion
        LOGGER.warn(f"Some queries are missing in the products file: {' '.join(missing_queries)}")
```
?

Comment (rnetser) [reply qe_metrics/utils/general.py:50]: https://github.com/RedHatQE/qe-metrics/pull/105/files#diff-22de0f516ff4af2d5cff55546a3b163c0bbef2a3aac88ed26c8154e9576bc7aaR52-R53 are now redundant

**Review by rnetser — DISMISSED** [review]

File: tests/test_utils/test_general.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def valid_queries():
 
 @pytest.fixture
```
Comment (myakove) [inline tests/test_utils/test_general.py:20]: please add test for 1 missing query as well

**Review by myakove — CHANGES_REQUESTED** [review]

File: qe_metrics/utils/general.py [hunk 3]
```diff
@@ -42,10 +42,14 @@ def verify_queries(queries_dict: dict[str, str]) -> None:
...
-        raise ValueError(f"Missing queries in the products file: {' '.join(missing_queries)}")
+    missing_queries = [query for query in required_queries if query not in queries_dict]
 
-    if none_values_queries := [query for query in required_queries if queries_dict.get(query) is None]:
+    if len(missing_queries) == len(required_queries):
+        raise ValueError(f"All queries are missing in the products file: {' '.join(missing_queries)}")
+    elif missing_queries:
+        LOGGER.warning(f"Some queries are missing in the products file: {' '.join(missing_queries)}")
+
+    if none_values_queries := [query for query in queries_dict if queries_dict.get(query) is None]:
```
Comment (myakove) [inline qe_metrics/utils/general.py:52]: `queries_dict.get(query) is None` empty string is a valid query?

Comment (myakove) [reply qe_metrics/utils/general.py:52]: sorry reading as  `is not None` :)

**Review by myakove — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**